src/CUDA_Runtime_Discovery.jl

```diff
@@ -265,6 +265,41 @@ function has_version_like_name(path)
     return true
 end
 
+function get_version_preference()
+    prefs = get(Base.get_preferences(), "CUDA_Runtime_jll", nothing)
```

Comment:
We can just depend on Preferences.jl here.

Reply:
What's the advantage?

Reply:
That it's not internal?

Reply:
And how, actually? I've never use Preferences.jl to check preferences of a different package. My first attempts failed:

```julia-repl
julia> get(Base.get_preferences(), "CUDA_Runtime_jll", nothing)
Dict{String, Any} with 2 entries:
  "local"   => "true"
  "version" => "12.1"

julia> has_preference(UUID("76a88914-d11a-5bdc-97e0-2f5a05c973a2"), "version")
false

julia> has_preference("CUDA_Runtime_jll", "version")
ERROR: ArgumentError: Cannot resolve package 'CUDA_Runtime_jll' in load path; have you added the package as a top-level dependency?
Stacktrace:
 [1] package_lookup_error(name::String)
   @ Preferences /projects/julia/.julia/packages/Preferences/enLcN/src/utils.jl:88
 [2] has_preference(name::String, key::String)
   @ Preferences /projects/julia/.julia/packages/Preferences/enLcN/src/Preferences.jl:76
 [3] top-level scope
   @ REPL[12]:1
```

Reply:
https://github.com/JuliaGPU/CUDA.jl/blob/bbe625bbd92cf2c3a5fde6aec1c940f0a1e2b039/src/utilities.jl#L93

Reply:
And yeah, I'd rather rely on the package that's intended for this use case, than the internal Base API that could change at any point.

Reply:
> [JuliaGPU/CUDA.jl@`bbe625b`/src/utilities.jl#L93](https://github.com/JuliaGPU/CUDA.jl/blob/bbe625bbd92cf2c3a5fde6aec1c940f0a1e2b039/src/utilities.jl#L93)

But the `CUDA_Runtime_jll` module isn't available here (in `CUDA_Runtime_Discovery.jl`). Should we also add `CUDA_Runtime_jll` as a dependency?

Reply:
Maybe hard-code the UUID then and use https://github.com/JuliaPackaging/Preferences.jl/blob/209a45060b8eef13a790300189dfd30c2cb6a1e1/src/Preferences.jl#L28-L35?

Reply:
Well, that's what I tried above and it only works if the package is on `LOAD_PATH`:

Not on `LOAD_PATH`, i.e. not in `Project.toml` here:

```julia-repl
julia> Base.get_preferences()
Dict{String, Any} with 2 entries:
  "MPIPreferences"   => Dict{String, Any}("abi"=>"OpenMPI", "_format"=>"1.0", "libmpi"=>"libmpi", "preloads"=>Union{}[], "mp…
  "CUDA_Runtime_jll" => Dict{String, Any}("local"=>"true", "version"=>"12.1")

julia> Base.get_preferences(UUID("76a88914-d11a-5bdc-97e0-2f5a05c973a2")) # UUID of CUDA_Runtime_jll
Dict{String, Any}()
```

With `CUDA_Runtime_jll` in `Project.toml`:

```julia-repl
julia> Base.get_preferences(UUID("76a88914-d11a-5bdc-97e0-2f5a05c973a2"))
Dict{String, Any} with 2 entries:
  "local"   => "true"
  "version" => "12.1"
```

Reply:
Note that the docstring of `load_preferences` also says
> loading preferences from all environments that list the given UUID as a direct dependency.

Reply:
The way I see it, we have two options:
1) Live with relying on internals (`Base.get_preferences`)
2) Drop the entire preferences story and only use the "parsing approach".

Reply:
Ah, I see. Let's just stick with the internal API then, for simplicity.